SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1261,14 +1298,26 @@ void USpatialNetDriver::ServerReplicateActors_ProcessPrioritizedActors(UNetConne
 	int32 ActorUpdatesThisConnection = 0;
 	int32 ActorUpdatesThisConnectionSent = 0;
 
+	const int32 NumActorsMigrating = MigrationHandler.GetActorsToMigrate().Num();
+
 	// SpatialGDK - Entity creation rate limiting based on config value.
 	uint32 EntityCreationRateLimit = GetDefault<USpatialGDKSettings>()->EntityCreationRateLimit;
 	int32 MaxEntitiesToCreate = (EntityCreationRateLimit > 0) ? EntityCreationRateLimit : INT32_MAX;
+	if (MaxEntitiesToCreate < NumActorsMigrating)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("EntityCreationRateLimit of %i ignored because %i actors need to migrate"), EntityCreationRateLimit, NumActorsMigrating);
+		MaxEntitiesToCreate = NumActorsMigrating;
+	}
 	int32 FinalCreationCount = 0;
 
 	// SpatialGDK - Actor replication rate limiting based on config value.
 	uint32 ActorReplicationRateLimit = GetDefault<USpatialGDKSettings>()->ActorReplicationRateLimit;
 	int32 MaxActorsToReplicate = (ActorReplicationRateLimit > 0) ? ActorReplicationRateLimit : INT32_MAX;
+	if (MaxActorsToReplicate < NumActorsMigrating)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("ActorReplicationRateLimit of %i ignored because %i actors need to migrate"), MaxActorsToReplicate, NumActorsMigrating);
```

Comment:
I agree it's not ideal, but this is just a (bad) performance tool to prevent general CPU usage from spending too much time replicating actors. It seems rare that the number of actors we want to migrate is ever larger than `ActorReplicationRateLimit` anyway, but in the off-chance it is, this seems like a reasonable solution - warn and increase the limit to ensure we can migrate all actors in the same frame.

Reply:
How does that limit cpu usage if we just warn and increase when we exceed it?
Unless that's why we don't like this anyway. Either way it seems like it should probably be discussed elsewhere.